doc/api/url.md

```diff
@@ -1847,7 +1847,25 @@ A `URIError` is thrown if the `auth` property is present but cannot be decoded.
 strings. It is prone to security issues such as [host name spoofing][]
 and incorrect handling of usernames and passwords. Do not use with untrusted
 input. CVEs are not issued for `url.parse()` vulnerabilities. Use the
-[WHATWG URL][] API instead.
+[WHATWG URL][] API instead, for example:
+
+```js
+function getURL(req) {
+  const proto = req.headers['x-forwarded-proto'] || 'https';
+  const host = req.headers['x-forwarded-host'] || req.headers.host || 'example.com';
+  return new URL(req.url || '/', `${proto}://${host}`);
```

Comment:
If `req.url` starts with `//`, the host will be overridden.
```js
new URL("//myhost.example.com/somepath/", "http://nodejs.org").href // http://myhost.example.com/somepath/
new URL("//myhost.example.com/somepath/", "http://nodejs.org").pathname // /somepath/
```
``new URL(`${proto}://${host}${req.url}`)`` would solve this problem, but I'm not sure if that's safe. As far as I tested, the `node:http` server rejected requests with a request target that doesn't start with `/` and `req.url` always started with `/`.

Reply:
Yes, `req.url` can start with `//`. Here's an example script:
```js
import http from 'node:http'

const server = http.createServer((req, res) => {
  console.log(`req.url: ${req.url}`)
  res.end('Hello, world!')
})
await new Promise((resolve) => {
  server.listen(3000, 'localhost', () => {
    resolve()
  })
})
const res = await fetch('http://localhost:3000//myhost.example.com/somepath/')
await res.text()
server.close()
```